luigi_pipeline/tests/test_seqr_gcnv_loading.py

```diff
@@ -0,0 +1,386 @@
+        )
+        genotype_task = SeqrGCNVGenotypesMTTask(
+            genome_version="38",
+            source_paths="i am completely ignored",
+            dest_path=self._genotypes_mt_file,
+            is_new_joint_call=True,
+        )
+        SeqrGCNVGenotypesMTTask.requires = lambda self: [variant_task]
+        worker.add(genotype_task)
+        worker.run()
+
+        variant_mt = hl.read_matrix_table(self._variant_mt_file)
+        self.assertEqual(variant_mt.count(), (3, 4))
+
+        key_dropped_variant_mt = variant_mt.rows().flatten().drop("variant_name", "svtype")
+        data = key_dropped_variant_mt.collect()
+        self.assertCountEqual(data, NEW_JOINT_CALLED_EXPECTED_VARIANT_DATA)
+
+        # Genotypes (only) Assertions
+        genotypes_mt = hl.read_matrix_table(self._genotypes_mt_file)
+        self.assertEqual(genotypes_mt.count(), (3, 4))
+
+        # Now mimic the join in BaseMTToESOptimizedTask
+        genotypes_mt = genotypes_mt.drop(*[k for k in genotypes_mt.globals.keys()])
+        row_ht = genotypes_mt.rows().join(variant_mt.rows()).flatten().drop("variant_name", "svtype")
+        data = row_ht.collect()
+        for i, row in enumerate(data):
+            data[i] = prune_empties(row)
+        self.assertListEqual(data, NEW_JOINT_CALLED_EXPECTED_VARIANT_AND_GENOTYPES_DATA)
+        
+
+    @mock.patch('lib.model.gcnv_mt_schema.datetime', wraps=datetime)
+    def test_run_merged_tsv_task(self, mock_datetime):
+        mock_datetime.date.today.return_value = datetime.date(2022, 12, 2)
+        worker = luigi.worker.Worker()
+        variant_task = SeqrGCNVVariantMTTask(
+            source_paths=MERGED_CALLSET,
+            dest_path=self._variant_mt_file,
+        )
+        genotype_task = SeqrGCNVGenotypesMTTask(
+            genome_version="38",
+            source_paths="i am completely ignored",
+            dest_path=self._genotypes_mt_file,
+            is_new_joint_call=False,
+        )
+        SeqrGCNVGenotypesMTTask.requires = lambda self: [variant_task]
+        worker.add(genotype_task)
+        worker.run()
+
+        variant_mt = hl.read_matrix_table(self._variant_mt_file)
+        self.assertEqual(variant_mt.count(), (2, 3))
+
+        key_dropped_variant_mt = variant_mt.rows().flatten().drop("variant_name", "svtype")
+        data = key_dropped_variant_mt.collect()
+        self.assertCountEqual(data, MERGED_EXPECTED_VARIANT_DATA)
+
+        genotypes_mt = hl.read_matrix_table(self._genotypes_mt_file)
+        self.assertEqual(genotypes_mt.count(), (2, 3))
+
+        # Now mimic the join in BaseMTToESOptimizedTask
+        genotypes_mt = genotypes_mt.drop(*[k for k in genotypes_mt.globals.keys()])
+        row_ht = genotypes_mt.rows().join(variant_mt.rows()).flatten().drop("variant_name", "svtype")
+        data = row_ht.collect()
+        for i, row in enumerate(data):
+            data[i] = prune_empties(row)
+        self.assertListEqual(data, MERGED_EXPECTED_VARIANT_AND_GENOTYPES_DATA)
```

Comment:
You may also want to test the `disabled` fields.

Reply:
You don't test the `SeqrGCNVMTToESTask`. That's why the [disabled fields](https://github.com/broadinstitute/seqr-loading-pipelines/blob/c794aff1f9f54094416db16b6c4fb92e328d7d3b/luigi_pipeline/seqr_loading_optimized.py#L77-L78) can't be tested.

Reply:
Yep, I'm going to take a shot at doing what you suggested regarding mocking closer to elastic search.  I initially left testing the elastic search export as out of scope because I don't really touch anything there too closely, but there's definitely major holes in these tests if we don't test `SeqrGCNVMTToESTask`.

Reply:
I will say, we are going to be sunsetting elasticsearch over the next few months. So while it is probably be a good idea to update the mocking to better test some elasticsearch behavior such as disable_index fields, I would recommend you not put too much time into it

Reply:
☝️ This was my initial thought.  I _think_ it's just a couple of lines.  If it takes more than that I'll reverse course.

Reply:
Got this working!  It was a _little_ more painful than expected (two tests pointing at the same mock 🤦‍♂️) but was definitely a worthy exercise since it found two nasty bugs.